apps/cli/src/executeMainThread.ts

```diff
@@ -192,9 +197,27 @@ export const executeMainThread = async () => {
 		tarService,
 	);
 
-	if (String(argv._) === "list") {
+	if (argv._.at(0) === "list" || argv._.at(0) === "search") {
 		try {
-			await handleListNamesCommand(printer, false);
+			const lastArgument = argv._.length > 1 ? String(argv._.at(-1)) : null;
+
+			let searchTerm: string | null = null;
+			if (lastArgument) {
+				if (lastArgument.length < 2) {
+					printer.printOperationMessage({
+						kind: "error",
+						message:
+							"Search term must be at least 2 characters long. Aborting...",
+					});
+					return;
+				}
+				searchTerm = lastArgument;
+			}
+
+			await handleListNamesCommand({
+				printer,
+				name: searchTerm ?? undefined,
```

Comment:
is search case sensitive? should we do searchTerm.toLowerCase() before searching?

Reply:
also, lets do searchTerm.trim()